src/Workspaces/Core/Portable/SymbolKey/SymbolKey.cs

```diff
@@ -75,114 +74,140 @@ namespace Microsoft.CodeAnalysis
     ///     </code>
     ///     The SymbolKey for both 'M' methods will be the same.  The SymbolKey will then resolve to both methods.
     /// </para>
+    /// <para>
+    /// <see cref="SymbolKey"/>s are not guaranteed to work across different versions of Roslyn.
+    /// They can be persisted in their <see cref="ToString()"/> form and used across sessions with
+    /// the same version of Roslyn. However, future versions may change the encoded format and may
+    /// no longer be able to <see cref="Resolve(Compilation, bool, bool, CancellationToken)"/> previous
+    /// keys.  As such, only persist if using for a cache that can be regenerated if necessary.
     /// </para>
     /// </summary>
     internal partial struct SymbolKey
     {
-        private readonly static Func<ITypeSymbol, bool> s_typeIsNull = t => t == null;
+        /// <summary>
+        /// Current format version.  Any time we change anything about our format, we should
+        /// change this.  This will help us detect and reject any cases where a person serializes
+        /// out a SymbolKey from a previous version of Roslyn and then attempt to use it in a 
+        /// newer version where the encoding has changed.
+        /// </summary>
+        private const int FormatVersion = 1;
 
         private readonly string _symbolKeyData;
 
-        public SymbolKey(string symbolKeyData)
+        /// <summary>
+        /// Constructs a new <see cref="SymbolKey"/> using the result of a previous call to
+        /// <see cref="ToString()"/> from this same session.  Instantiating with a string 
+        /// from any other source is not supported.
+        /// </summary>
+        public SymbolKey(string data)
         {
-            _symbolKeyData = symbolKeyData ?? throw new ArgumentNullException();
+            _symbolKeyData = data ?? throw new ArgumentNullException();
         }
 
-        public static IEqualityComparer<SymbolKey> GetComparer(bool ignoreCase, bool ignoreAssemblyKeys)
+        /// <summary>
+        /// Constructs a new <see cref="SymbolKey"/> representing the provided <paramref name="symbol"/>.
+        /// </summary>
+        public SymbolKey(ISymbol symbol, CancellationToken cancellationToken = default)
```

Comment:
Feels weird to take a cancellation token in a ctor. Maybe just remove this and always use the Create method?

Reply:
sounds good.